File: PlantsAPI/Models/Plant.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace PlantsAPI.Models
{
    public class Plant
    {
        public Guid Id { get; set; }

        [MaxLength(50)]
        public string Name { get; set; }

       
        public string? Description { get; set; } 
        public string? ImageUrl { get; set; }

        public string? Note { get; set; }
        public int? Interval { get; set; }
        public DateTime? LastNotification { get; set; }
        public DateTime? NextNotification { get; set; }

        public Guid UserId { get; set; }

        [JsonIgnore]
        public virtual User? User { get; set; }
    }
}

File: PlantsWeb/src/app/models/Plant.ts
import { SafeUrl } from "@angular/platform-browser";
import { User } from "./User";

export class Plant {

    constructor(
        //Plant data
        public id : string,
        public name : string,
        public description : string | null,
        public imageData : string | null | SafeUrl,

        //Maintenance data
        public note : string | null,
        public interval : number | null,
        public lastNotification: Date | null,
        public nextNotification: Date | null,
        
        //User info
        public user : User | null,
        public userId: string){
    }
}